ide/commons-gwt/src/main/java/org/eclipse/che/ide/resource/Path.java

```diff
@@ -1143,4 +1176,24 @@ public static Path commonPath(Path... paths) {
 
     return commonPath;
   }
+
+  /** Returns path encoded by segments without device. */
+  public String getEncodedPath() {
```

Comment:
Is it tested?

Create separate utility class called `PathEncoder` and move this method as static there. See how `java.net.URLEncoder#encode(java.lang.String, java.lang.String)` method done.

This is not a right place for this method, it should be a static utility method.

Reply:
Yes it was tested.